grp/classic.gd

```diff
@@ -193,13 +192,17 @@ DeclareConstructor( "GeneralOrthogonalGroupCons",
 
 #############################################################################
 ##
-#F  GeneralOrthogonalGroup( [<filt>, ][<e>, ]<d>, <q> ) .  gen. orthog. group
-#F  GO( [<filt>, ][<e>, ]<d>, <q> )
+#F  GeneralOrthogonalGroup( [<filt>, ][<e>, ]<d>, <q>[, <form>] )
```

Comment:
But doesn't the form already imply the value for `d` and `e`?  If we accept all three, how are they supposed to interact / be sanity checked?

I guess only the base field would perhaps we in need of overriding (i.e., it seems reasonably to say: "use this form given over GF(2) but create a group over GF(8)").

Reply:
My idea was the following:
One can write `GO( 3, 5, form )` in order to express that one wants the group in question, with the given form as an additional requirement. One can also write `GO( form )` (which is also a documented variant), if one wants to express that the given form determines the other parameters.
In code examples, I would probably prefer the former variant.

Reply:
But if the former is done, shouldn't we then also check that the `form` matches the given values for `d` and `q`? Or do we expect all installed methods to re-implement that check?

Reply:
The GAP library does not know about form objects, and the code dealing with this argument belongs to the Forms package.
We could check some conditions about a matrix or a group with stored form in the global functions in the GAP library, but only the code in the Forms package can decide whether the given argument makes sense.